support/utils/functions_test.go

```diff
@@ -36,3 +36,90 @@ func TestDedupe(t *testing.T) {
 		})
 	}
 }
+
+func TestToMapStringInterface_SuccessMap(t *testing.T) {
+	success := map[string]interface{}{
+		"test": true,
+	}
+	m, e := ToMapStringInterface(success)
+	assert.Equal(t, success, m)
+	assert.NoError(t, e)
+}
+
+func TestToMapStringInterface_SuccessStruct(t *testing.T) {
+	type NonEmptyStruct struct {
+		Value int `json:"value"`
+	}
+
+	type SuccessStruct struct {
+		True           bool
+		False          bool
+		EmptyString    string
+		NonEmptyString string
+		Int            int
+		Float          float64
+		EmptyStruct    struct{}
+		NonEmptyStruct NonEmptyStruct
+		EmptyInts      []int
+		NonEmptyInts   []int
+	}
+
+	success := SuccessStruct{
+		True:           true,
+		False:          false,
+		EmptyString:    "",
+		NonEmptyString: "hello world",
+		Int:            1,
+		Float:          1.0,
+		EmptyStruct:    struct{}{},
+		NonEmptyStruct: NonEmptyStruct{1},
+		EmptyInts:      []int{},
+		NonEmptyInts:   []int{1, 2},
+	}
+
+	want := map[string]interface{}{
+		"EmptyInts":      []interface{}{},
+		"EmptyString":    "",
+		"EmptyStruct":    map[string]interface{}{},
+		"False":          false,
+		"Float":          1.,
```

Comment:
I know go allows this, but can we avoid it and use 1.0 instead for all places where we do this

there are a few such things that are syntactic sugar, but which make the codebase more complex for newcomers and non-go developers, so best to avoid where possible